cpp/basix/version.h.cmake

```diff
@@ -0,0 +1,10 @@
+// Copyright (c) 2020 Chris Richardson
```

Comment:
I'm confused by the name of this file. Doesn't look like a CMake file. Should it be `version.h.in` (a template)?

A possibly simpler approach is to just put the versions numbers in `version.h` and let CMake parse the versions, see https://github.com/xtensor-stack/xtensor/blob/master/CMakeLists.txt#L15.